### PR TITLE
Resolved warnings

### PR DESCRIPTION
**Pull request by RoiEXLab**
This should resolve a lot of warnings in the current master

Removed the Deprecated annotation from MD5Crypt, since we may need to keep it to enable users to still log in.

EDIT: The non-test javafiles are no longer using deprecated methods or classes

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/969)

### CODE REVIEW

**Comment by RoiEXLab** [comment]
@DanVanAtta sorry, this doubles #940 at some point

**Comment by RoiEXLab** [comment]
Still, since #930 has merge conflocts, this PR should be merged before resolving those.

**Comment by RoiEXLab** [comment]
@ron-murhammer if you have the time please review this one please... 98% refactoring

File: src/games/strategy/engine/framework/GameRunner2.java [hunk 1]
```diff
@@ -137,16 +138,15 @@ private static void usage() {
...
-    }
-    (new Thread(() -> setupLogging())).start();
-
+    PerfTimer timer = PerfTimer.startTimer("Show main window");
+    // do after we handle command line args
+    checkForMemoryXMX();
+    SwingUtilities.invokeLater(() -> setupLookAndFeel());
+    showMainFrame();
+    new Thread(() -> setupLogging()).start();
+    timer.stop();
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner2.java]: Why add in the extra line of time.stop()? The try-with-resource block would call 'close()' automatically, stopping the timer.

Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner2.java]: In addition, the timer is being changed, now includes the time to spawn a new thread. Not consequential, but we should be careful of functionality changes. 
Bottom line though, the timer is more useful I think for spot checking timings, rather than for the metrics reporting it was originally intended for, so the main window timing can be removed altogether.

File: src/games/strategy/engine/framework/map/download/DownloadFile.java [hunk 2]
```diff
@@ -68,14 +67,14 @@ public void startAsyncDownload() {
...
     return new Thread(() -> {
       if (state != DownloadState.CANCELLED) {
-        try (PerfTimer timer = Perf.startTimer("Download map: " + downloadDescription.getUrl())) {
-          URL url = downloadDescription.newURL();
-          try {
-            DownloadUtils.downloadFile(url, fileToDownloadTo);
-          } catch (Exception e) {
-            ClientLogger.logError("Failed to download: " + url, e);
-          }
+        PerfTimer timer = PerfTimer.startTimer("Download map: " + downloadDescription.getUrl());
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/map/download/DownloadFile.java]: This timer perhaps should remain IMO. But again, why remove the try-with block?

File: src/games/strategy/engine/history/History.java [hunk 3]
```diff
@@ -28,6 +28,8 @@
 import games.strategy.triplea.ui.history.HistoryPanel;
 
 public class History extends DefaultTreeModel {
+  private static final long serialVersionUID = -1769876896869L;
```
Comment (DanVanAtta) [inline src/games/strategy/engine/history/History.java]: Please try to split up independent changes into different PRs. Any objections is going to hold up all work in this PR. In addition it is difficult to review when the diff is really long, and when there are many different changes.

File: src/games/strategy/engine/framework/GameRunner2.java (hunk 1 above)
Comment (RoiEXLab) [inline src/games/strategy/engine/framework/GameRunner2.java]: I know, and that was the problem: The Java Compiler dislikes the fact the try with resource statement declares a variable not being used inside the block

File: src/games/strategy/engine/message/HasEndPointImplementor.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package games.strategy.engine.message;
```
Comment (DanVanAtta) [inline src/games/strategy/engine/message/HasEndPointImplementor.java]: Yeah, this is not fun to review. When breaking out the inner classes, that should be the *only* thing in the PR. It is not clear what has changed.

File: src/games/strategy/engine/framework/GameRunner2.java (hunk 1 above)
Comment (RoiEXLab) [inline src/games/strategy/engine/framework/GameRunner2.java]: And that thing with the thread was unintended. whoops

**Comment by RoiEXLab** [comment]
@DanVanAtta my goal was to get rid of all remaining warnings... Mostly refactoring changes, sorry if the "by the way changes" were too much

**Comment by DanVanAtta** [comment]
There appear to be at least 3 or 4 categories of change in this PR. Please try to break things up into smaller chunks. There is already lots of opportunity for various conversation on multiple topics, which will add back and forth time, meanwhile the other changes are idle. Changes sitting on a branch is a real problem, represents overhead. Regardless, if you would, break this PR up, otherwise will try to get back to review soon'ish, only about 50% way through here.

File: src/games/strategy/engine/framework/GameRunner2.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner2.java]: The variable is needed so java has something to call close on. Ideally we would not even need that. The try/with block is less error prone, there is no way to forget to call the 'close'. 

But, also, there is another functional change, the timer.close() is no longer called in a finally. If the timed operation were to crash, the timer would never be stopped.

**Comment by DanVanAtta** [comment]
@RoiEXLab  with respect to merge ordering, @ron-murhammer and I had a few conversations about that. I asked that we merge from oldest to newest whenever possible. This makes merge conflicts more predictable. At that time we had a few PRs that were large and old, and new PRs were being merged in before. In the extreme example the PR had 7 different rounds of merge conflict before it was reviewed and pulled in. So the idea is that this PR, as long as it does not conflict with any of the earlier PRs, then it won't be in conflict with any new upcoming PRs. If you suspect #930 will be merged, and I expect for that to happen, I recommend you rebase this PR on #930.

File: src/games/strategy/engine/framework/GameRunner.java [hunk 5]
```diff
@@ -34,28 +33,21 @@ private static void checkJavaVersion() {
...
+      JOptionPane.showMessageDialog(null,
+          "TripleA requires a java runtime greater than or equal to 8 [ie: Java 8]. \nPlease download a newer version of java from http://www.java.com/",
+          "ERROR", JOptionPane.ERROR_MESSAGE);
+      System.exit(-1);
     }
-  }// end checkJavaVersion()
+  }
 
   public static void main(final String[] args) {
+    PerfTimer timer = PerfTimer.startTimer("GameRunner1 Total Launch");
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner.java]: IMO, let's remove any usages of PerfTimer for the moment. Keep things simpler, and it does seem to me that it looks like this timer is best for spot checking, or on-the-spot performance investigation. If this were server side code of some sort, we would want it, but since we have a thick client, unless we are going to report these values back to a centralized place, it is not necessary.

File: src/games/strategy/engine/framework/GameRunner.java [hunk 6]
```diff
@@ -34,28 +33,21 @@ private static void checkJavaVersion() {
...
     // we want this class to be executable in older jvm's
-    // since we require jdk 1.5, this class delegates to GameRunner2
+    // since we require jdk 1.8, this class delegates to GameRunner2
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner.java]: The original comment does not make a lot of sense to me. Why do we delegate to GameRunner2 becasue we require jdk1.5? 
IMO the comment is really just rephrasing the code, `checkJavaVersion`, which presumably will crash or something if the wrong java version is here. I recommend just removing this comment, it is not very helfpul.

File: src/games/strategy/engine/framework/GameRunner2.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner2.java]: As mentioned above, let's remove the timer, this way you don't have to worry about the try/finally idiom.

File: src/games/strategy/engine/framework/map/download/DownloadFile.java (hunk 2 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/map/download/DownloadFile.java]: AS mentioned, let's remove the timer.

File: src/games/strategy/engine/history/SerializationWriter.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package games.strategy.engine.history;
+
+import java.io.Serializable;
+
+public interface SerializationWriter extends Serializable {
```
Comment (DanVanAtta) [inline src/games/strategy/engine/history/SerializationWriter.java]: Can we reduce teh scope of this interface? Was this actually causing warnings? A small 'inner' interface is not necessarily a bad thing. For classes with logic, their own class file I think is almost always good, but here, not necessarily convinced.

Failing the above, adding in a class javadoc to describe how this interface is used, or should be used, woudl be of help.

**Comment by RoiEXLab** [comment]
@DanVanAtta breaking up this PR into #973 , #974 and #975

File: src/games/strategy/engine/history/SerializationWriter.java (hunk 7 above)
Comment (RoiEXLab) [inline src/games/strategy/engine/history/SerializationWriter.java]: @DanVanAtta Java dislikes the fact that auxiliary classes are used in the same package.
If this interface would only be used inside History.java this wouldn't be a problem

File: src/games/strategy/performance/EnablePerformanceLoggingCheckBox.java [hunk 8]
```diff
@@ -10,12 +10,12 @@
 
   public EnablePerformanceLoggingCheckBox() {
     super("Enable Performance Logging");
-    setSelected(Perf.isEnabled());
+    setSelected(PerfTimer.isEnabled());
```
Comment (DanVanAtta) [inline src/games/strategy/performance/EnablePerformanceLoggingCheckBox.java]: Are you sure about this update here? I know one weakness of the timer classes was Perf probably should have been called PerfTimer, but regardless, Perf is the only thing that should be touching PerfTimer. You get a PerfTimer from Perf, and then the auto-close try/catch blocks closes the PerfTimer. In fact we probably should make all methods in PerfTimer private/protected except for close() to enforce that relationship.

File: src/games/strategy/performance/PerfTimer.java [hunk 9]
```diff
@@ -1,8 +1,12 @@
 package games.strategy.performance;
 
-import java.io.Closeable;
+import java.util.prefs.Preferences;
 
-public class PerfTimer implements Closeable {
```
Comment (DanVanAtta) [inline src/games/strategy/performance/PerfTimer.java]: No, we want this to be Closeable. Whenever we use a timer, it should be closed in a 'finally' block to make sure we always close the timer. There was a reason for there being two classes, Perf provided access to a PerfTimer, which was then auto-closeable.

If you could not tell,I do not agree with the combination of the two classes. The role of one class was a factory, the other one is a timer.

**Comment by DanVanAtta** [comment]
Thanks for the break-up @RoiEXLab. I just finished reviewing this, the split will make things easier IMO for us all. Please take a look at my last few comments and apply them forward if you would.

File: src/games/strategy/engine/framework/GameRunner.java (hunk 6 above)
Comment (RoiEXLab) [inline src/games/strategy/engine/framework/GameRunner.java]: @DanVanAtta I think originally TripleA was designed to even start with older Java versions and if the java version is not high enough to crash with the message "You need to update your java to \[Version]\"
